src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -507,5 +507,26 @@ public static int countEmptyStrings(String... strings) {
      */
     public static String convertToEmptyStringIfNull(String str) {
         return (str == null) ? "" : str;
-    }  
+    }
+    
+    /**
+     * Removes the outermost enclosing square brackets surrounding a string.
+     * 
+     * @param str
+     * @return the string without the outermost enclosing square brackets and 
+     *         without leading and trailing spaces, if the given string is enclosed by square brackets <br/> 
+     *         the string itself if the given string is not enclosed by square brackets <br/>
+     *         null if the given string is null
+     */
```

Comment:
The trimming is done in the previous method (removeBracketsForArrayString). I would think pre-trimming the input string before we remove the brackets are needed to handle the case when the input string has leading and trailing spaces (eg. ```"  [string]  "```). 
But, as for post-trimming, I agree we should let the caller of the method do the trimming.